clang/lib/CodeGen/CGVTables.cpp

```diff
@@ -831,9 +831,16 @@ void CodeGenVTables::addVTableComponent(ConstantArrayBuilder &builder,
 
     // Pure virtual member functions.
     if (cast<CXXMethodDecl>(GD.getDecl())->isPureVirtual()) {
-      if (!PureVirtualFn)
-        PureVirtualFn =
-            getSpecialVirtualFn(CGM.getCXXABI().GetPureVirtualCallName());
+      if (!PureVirtualFn) {
```

Comment:
Should we just add this to `getSpecialVirtualFn` ?

Reply:
So, on the one hand, it would also cover deleted virtual functions. On the other hand, the attributes (in form of SYCL properties) for virtual functions that are callable from device cannot be applied to deleted virtual functions so it won't bring any extra benefit.

I think that the code looks cleaner if we move this change. Even though it is now a little bit more complicated to trace what are the special functions which we are ignoring for SYCL device code (because the filtering is within a helper lambda and not directly within an `if pure virtual`), the code overall seems cleaner to me anyway (less new `{ }`).

Moved it in 6eeb45f6cfe76f0e22017135fc63048639d9c3b4